modules/activiti-engine/src/main/java/org/activiti/engine/impl/history/DefaultHistoryManager.java

```diff
@@ -389,6 +390,15 @@ public void recordExecutionReplacedBy(ExecutionEntity execution, InterpretableEx
       for (HistoricActivityInstanceEntity historicActivityInstance: historicActivityInstances) {
         historicActivityInstance.setExecutionId(replacedBy.getId());
       }
+
+      // Update the persisted historic task instances that are open
```

Comment:
Is not the same fix as HistoricActivityInstanceEntity's cache for HistoricTaskInstanceEntity's cache unnecessary?
> // Update the cached historic activity instances that are open

Reply:
Here, I am trying to update the executionId of the unfinished(active) Task Instances when needed to replace in the Concurrent execution. As if user tries to view the active Task Instance form, It fetches the variables by referring  to the executionId from the hi_taskinst table.

Reply:
I understand your use case.
But if you will fix this data inconsistency problem concerning executionId, you should make consistent fix for this problem.

In existing implementations related to HistoricActivityInstanceEntity, the executionId is replaced including caches. https://github.com/lourdupranay/Activiti/blob/f5269ec807aeca5c198d8277be24c136ea9a2284/modules/activiti-engine/src/main/java/org/activiti/engine/impl/history/DefaultHistoryManager.java#L335

In addition, you fix only about act_hi_taskinst table, but I think the same problem exists in the following tables.
* act_hi_varinst
* act_hi_detail
* act_evt_log

Reply:
@daisuke-yoshimoto I'm not sure about updating act_hi_detail and acti_evt_log. Historic data shouldn't be updated in general unless there is a running instance that changed.

Reply:
@balsarori 

> I'm not sure about updating act_hi_detail and acti_evt_log. 

I see. It's OK.

Then, in addition to the original fix for act_hi_taskinst table, is it OK to just fix for act_hi_varinst table in the same way?

Reply:
Hello,
Usually in act_hi_varinst table executionId is stored with the parent executionId of the related execution , It differs only when we have multi-instance activiti. But a gateway or event cannot become multi-instance . Hence , there will no need to replace executionId in act_hi_varinst table .

Reply:
Variables are execution specific (```execution.setVariable```) so, I don't think act_hi_varinst should be updated

Reply:
@balsarori @lourdupranay 
How do you think about task local variable?
For example, In the following cases, ExecutionId will differ between the act_ru_variable table and the act_hi_varinst table.

![task_local_var](https://user-images.githubusercontent.com/16251291/34108290-da37236a-e442-11e7-95ac-9501b92101b9.png)

Reply:
@daisuke-yoshimoto  Can you please let me know the engine version you are using ?

Reply:
@lourdupranay 
I use Activiti 5.22.0 for your review.

Reply:
Good point @daisuke-yoshimoto although I don't think that execution id is important for local tasks variables. 

BTW I like the screen shots seems that you are still using activiti-explorer :)

Reply:
@daisuke-yoshimoto  I am using 5.22.0.5 version and  i could not reproduce it.

Reply:
>  i could not reproduce it.

You can reproduce it by the following code and bpmn.

https://files.gitter.im/daisuke-yoshimoto/EUvh/user_task_local_variable.bpmn20.xml

```java
@Deployment
public void testTaskLocalVar() {
  runtimeService.startProcessInstanceByKey("user_task_local_variable").getId();
  
  List<Task> taskList = taskService.createTaskQuery().list();
  String oldExecutionId = taskList.get(1).getExecutionId();

  taskService.complete(taskList.get(0).getId());
  
  Task task = taskService.createTaskQuery().singleResult();
  String executionId = task.getExecutionId();
  
  HistoricVariableInstance var1 = historyService.createHistoricVariableInstanceQuery().executionId(executionId).singleResult();
  assertEquals(var1, null);

  HistoricVariableInstance var2 = historyService.createHistoricVariableInstanceQuery().executionId(oldExecutionId).singleResult();
  assertEquals(var2.getValue(), "AAA");
  
}
```

Reply:
@balsarori 

> although I don't think that execution id is important for local tasks variables.

I do not know whether it is important, but since it is inconsistent data problem, I want to fix it if possible. For example, The search result using HistoricVariableInstanceQuery.executionId (String) becomes incorrect.

> you are still using activiti-explorer

I use activiti-explorer to review this pull request because this is fix for 5.x.

Reply:
I agree with you, it should be fixed.

Reply:
@lourdupranay can you test against the 5.x branch?